src/extension.ts

```diff
@@ -1248,10 +1264,25 @@ export async function activate(context: vscode.ExtensionContext) {
       .on("core-dump-detected", (resp) => {
         // perform core-dump related stuff here
         //once done call .done() to notify the monitor process
+        const projectElfFile = "my-custom.elf";
+        const debugAdapterConfig = {
+          elfFile: projectElfFile,
+          isPostMortemDebugMode: false,
+        } as IDebugAdapterConfig;
+        debugAdapterManager.configureAdapter(debugAdapterConfig);
+        debugAdapterManager.start();
```

Comment:
This won't start the debugging programmatically, but rather we have to call 

```js
await vscode.debug.startDebugging(undefined, {
  name: "Core Dump Debug",
  type: "espidf",
  request: "launch",
});
```

which will launch the debug adapter by calling 

`vscode.debug.registerDebugAdapterDescriptorFactory("espidf", { ` __[Line:661](https://github.com/espressif/vscode-esp-idf-extension/pull/139/files?file-filters%5B%5D=.ts#diff-45327f86d4438556066de133327f4ca2R661)__ and there we have to see if its started by `Core Dump Debug` by testing `name` then can parse elf file and `core-dump` elf file